webpack/assets/javascripts/react_app/routes/Audits/AuditsPage/AuditsPage.js

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable camelcase */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Icon, Spinner } from 'patternfly-react';
+import './auditspage.scss';
+
+import { translate as __ } from '../../../common/I18n';
+import AuditsPageEmptyState from './AuditsPageEmptyState';
+import PageLayout from '../../common/PageLayout/PageLayout';
+import AuditsList from '../../../components/AuditsList';
+import Pagination from '../../../components/Pagination/Pagination';
+import { getParams } from '../../../components/Pagination/PaginationHelper';
+import {
+  AUDITS_DOC_URL,
+  AUDITS_SEARCH_PROPS,
+  AUDITS_PER_PAGE_OPTIONS,
+} from '../constants';
+
+class AuditsPage extends Component {
+  componentDidMount() {
+    document.title = __('Audits');
+    const { audits, initializeAudits } = this.props;
+    if (audits.length === 0) {
+      const params = getParams();
+      initializeAudits(params, true);
+    }
+
+    window.onpopstate = e => {
+      if (e.state) {
+        const popStateParams = getParams();
+        initializeAudits(popStateParams);
+      }
+    };
+  }
+
+  componentWillUnmount() {
+    window.onpopstate = () => {};
+  }
+
+  render() {
+    const {
+      audits,
+      page,
+      perPage,
+      searchQuery,
+      itemCount,
+      showMessage,
+      message,
+      auditSearch,
+      changePage,
+      changePerPage,
+      isFetchingNext,
+      isFetchingPrev,
+    } = this.props;
+    return (
+      <PageLayout
+        header={__('Audits')}
+        searchable
+        searchProps={AUDITS_SEARCH_PROPS}
+        searchQuery={searchQuery}
+        onSearch={auditSearch}
+        onBookmarkClick={auditSearch}
+        toolbarButtons={
+          <Button href={AUDITS_DOC_URL} className="btn-docs">
+            <Icon type="pf" name="help" />
+            {__(' Documentation')}
+          </Button>
+        }
+      >
+        {showMessage ? (
+          <AuditsPageEmptyState message={message} />
+        ) : (
+          <React.Fragment>
+            {audits.length === 0 ? (
+              <div id="loading-audits">
+                <Spinner size="lg" loading />
```

Comment:
Maybe [LoadingState](https://github.com/patternfly/patternfly-react/tree/master/packages/patternfly-3/patternfly-react/src/components/LoadingState) instead?

[Example use](https://github.com/theforeman/foreman_templates/pull/109/files#diff-786cc5f7ee55e022274bd6292c2837a7R28)

Reply:
LoadingState will hide the table when loading is true, we don't want that because it will flicker between pages.